blockchain/merkle.go

```diff
@@ -5,12 +5,42 @@
 package blockchain
 
 import (
+	"bytes"
+	"fmt"
 	"math"
 
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/txscript"
 	"github.com/btcsuite/btcutil"
 )
 
+const (
+	// CoinbaseWitnessDataLen is the required length of the only element within
+	// the coinbase's witness data if the coinbase transaction contains a
+	// witness commitment.
+	CoinbaseWitnessDataLen = 32
+
+	// CoinbaseWitnessPkScriptLength is the length of the public key script
+	// containing an OP_RETURN, the WitnessMagicBytes, and the witness
+	// commitment itself. In order to be a valid candidate for the output
+	// containing the witness commitment
+	CoinbaseWitnessPkScriptLength = 38
+)
+
+var (
+	// WitnessMagicBytes is the prefix marker within a public key script to
+	// indicate that this output holds the witness commitment for a block.
+	// The prefix breaks down down: OP_RETURN OP_DATA_36
```

Comment:
Modified the comment to mention that it's an output of the coinbase transaction, and removed the second portion of the comment.